octobot_trading/personal_data/exchange_personal_data.py

```diff
@@ -203,7 +205,11 @@ async def handle_order_update_from_raw(self, exchange_order_id, raw_order,
 
     async def update_order_from_stored_data(self, exchange_order_id, pending_groups):
         order = self.orders_manager.get_order(None, exchange_order_id=exchange_order_id)
+        previous_order_id = order.order_id
         await orders_storage_operations.apply_order_storage_details_if_any(order, self.exchange_manager, pending_groups)
+        if previous_order_id != order.order_id:
+            # order_id got restored to its original value
+            self.orders_manager.replace_order(previous_order_id, order)
```

Comment:
How the order id can change here?

Reply:
the initial order id when an order is fetched from exchange is a random uuid generated in constructor, we can't keep it when the order is related to a stored order

Reply:
If I understand well, when an order is loaded from the order database we need to restore the order id to match the one in the database right?

Reply:
Right, because this order_id is supposed to remain constant for a given order (mainly for trading signals where the order is identified by this id)

Reply:
Right, thanks :+1: